src/wp-includes/rest-api/endpoints/class-wp-rest-templates-controller.php

```diff
@@ -68,7 +68,12 @@ public function register_routes() {
 		// Lists/updates a single template based on the given id.
 		register_rest_route(
 			$this->namespace,
-			'/' . $this->rest_base . '/(?P<id>[\/\s%\w\.\(\)\[\]\@_\-]+)',
+			/**
+			 * ([^\/]+(?:\/[^\/]+)?) matches the theme's directory: `/themes/<subdirectory>/<theme>/` or `/themes/<theme>/`.
+			 * [\/\w-]+ matches the template name.
+			 * Note: `|` is an invalid character for a Windows' directory name.
+			 */
+			'/' . $this->rest_base . '/(?P<id>([^\/\|]+(?:\/[^\/\|]+)?)[\/\w-]+)',
```

Comment:
Before  [changeset 52376](https://core.trac.wordpress.org/changeset/52376), the regex was

```php
'/' . $this->rest_base . '/(?P<id>[\/\w-]+)',
```

The change restores the original regex but also adds theme matching, which is the same regex as in the global styles controller change above.